src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSender.java

```diff
@@ -15,177 +15,266 @@
 import com.exactpro.th2.common.schema.message.MessageSender;
 import com.exactpro.th2.common.schema.message.impl.rabbitmq.configuration.ResendMessageConfiguration;
 import com.exactpro.th2.common.schema.message.impl.rabbitmq.connection.ConnectionManager;
-import com.exactpro.th2.common.schema.message.impl.rabbitmq.connection.retry.CreateChannelException;
-import com.exactpro.th2.common.schema.message.impl.rabbitmq.connection.retry.RetryBuilder;
 import com.rabbitmq.client.Channel;
 import com.rabbitmq.client.ConfirmListener;
+import com.rabbitmq.client.ShutdownSignalException;
 import io.prometheus.client.Counter;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.jetbrains.annotations.NotNull;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.IOException;
+import java.util.Arrays;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicBoolean;
-import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.locks.Condition;
-import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
+import java.util.stream.Collectors;
 
-public abstract class AbstractRabbitSender<T> implements MessageSender<T>, ConfirmListener {
+import static java.util.Objects.requireNonNull;
+
+public abstract class AbstractRabbitSender<T> implements MessageSender<T> {
     private static final Logger LOGGER = LoggerFactory.getLogger(AbstractRabbitSender.class);
+    private static final AtomicLong SEND_DATA_REQUEST_COUNTER = new AtomicLong(0);
 
     private final ConnectionManager connectionManager;
     private final ResendMessageConfiguration resendMessageConfiguration;
-    private final String sendQueue;
+    private final String routingKey;
     private final String exchangeName;
-
-
-    private final Lock lock = new ReentrantLock();
-    private final Condition notLock = lock.newCondition();
-    private final AtomicBoolean canSend = new AtomicBoolean(true);
-    private final AtomicInteger countAttempts = new AtomicInteger(0);
-
-
-    private final Object channelLock = new Object();
+    private final HealthMetrics healthMetrics = new HealthMetrics(this); //TODO: Pass attempts count
+
+    /**
+     * Scheduled executor service with single thread. It provide ability to write task oriented code without concurrent affects
+     */
+    private final ScheduledExecutorService scheduler = new ScheduledThreadPoolExecutor(1);;
```

Comment:
Please use single ScheduledThreadPoolExecutor(1) in common factory for mq routers